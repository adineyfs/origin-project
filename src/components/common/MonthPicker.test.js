import React from "react";
import { shallow } from "enzyme";
import MonthPickerTest from "./MonthPicker";

describe("Month Picker", () => {
  it("Should render with default props", () => {
    const component = shallow(<MonthPickerTest />);
    expect(component).toMatchSnapshot();
  });

  it("Should render with custom props and call functions correctly", () => {
    const handleRightClickMock = jest.fn();
    const handleLeftClickMock = jest.fn();
    const component = shallow(
      <MonthPickerTest
        onForwardClick={handleRightClickMock}
        onBackClick={handleLeftClickMock}
        month="June"
        year={2021}
      />
    );
    component.find(".month-picker__arrow-button--left").simulate("click");
    expect(handleLeftClickMock).toHaveBeenCalledTimes(1);

    component.find(".month-picker__arrow-button--right").simulate("click");
    expect(handleRightClickMock).toHaveBeenCalledTimes(1);

    expect(component.find(".month-picker__month-label").text()).toBe("June");
    expect(component.find(".month-picker__year-label").text()).toBe("2021");
  });

  it("Should render with custom props and call functions correctly", () => {
    const handleRightClickMock = jest.fn();
    const handleLeftClickMock = jest.fn();
    const component = shallow(
      <MonthPickerTest
        onForwardClick={handleRightClickMock}
        onBackClick={handleLeftClickMock}
        month="June"
        year={2021}
      />
    );
    component.find(".month-picker__arrow-button--left").simulate("click");
    expect(handleLeftClickMock).toHaveBeenCalledTimes(1);

    component.find(".month-picker__arrow-button--right").simulate("click");
    expect(handleRightClickMock).toHaveBeenCalledTimes(1);

    expect(component.find(".month-picker__month-label").text()).toBe("June");
    expect(component.find(".month-picker__year-label").text()).toBe("2021");
  });
});
