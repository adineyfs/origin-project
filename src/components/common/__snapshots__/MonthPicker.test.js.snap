// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month Picker Should render with default props 1`] = `
<Fragment>
  <div
    className="month-picker "
  >
    <div
      className="month-picker__arrow-button month-picker__arrow-button--left"
      onClick={[Function]}
    >
      <Memo(ForwardRef(ArrowBackIosIcon))
        color="disabled"
      />
    </div>
    <div
      className="month-picker__month-label"
    />
    <div
      className="month-picker__year-label"
    >
      0
    </div>
    <div
      className="month-picker__arrow-button month-picker__arrow-button--right"
      onClick={[Function]}
    >
      <Memo(ForwardRef(ArrowForwardIosIcon))
        color="disabled"
      />
    </div>
  </div>
  <span
    className="month-picker__month-error-text-clear"
  >
    Only future dates
  </span>
</Fragment>
`;
